src/presentation/components/ItemCards.tsx

```diff
@@ -30,14 +30,25 @@ const ItemCards: React.FC = () => {
 
   useEffect(() => {
     //Resolves promise to set the data from JSON into the custom hook.
-    apiCallFunc().then(res => setData(res.items));
+    fetchItems()
+    .then(res => setData(res.items))
+    .catch(err => setData(err));
   },[setData])
-  console.log("length of data", data?.length);
-  console.log("show more", showMore);
+
+  //Loading state
+  if(data.length === 0) {
+      return(<h1> Loading Items... </h1>);
+  };
+
+  //error state
+  if(!Array.isArray(data)) {
+    return(<h1>Failed to load items</h1>)
+  }
+
   return (
     <div>
       <Grid direction="row" container>
-      {data && data.slice(0, showMore).map((item) => {
+      {Array.isArray(data) && data.slice(0, showMore).map((item) => {
```

Comment:
why are we slicing? we want to minimize logic in this file and put all logic in the applicatoin layer.

In this file, we should only be doing `data.map((item) => ...` 
the application layer should handle logic to show the items